rusk-recovery/src/state.rs

```diff
@@ -45,210 +45,198 @@ pub static FAUCET_KEY: Lazy<PublicSpendKey> = Lazy::new(|| {
 });
 
 fn generate_transfer_state(
+    session: &mut Session,
     snapshot: &Snapshot,
-    state: &mut NetworkState,
-) -> Result<TransferContract, Box<dyn Error>> {
-    let mut transfer: TransferContract =
-        state.get_contract_cast_state(&rusk_abi::transfer_contract())?;
+) -> Result<(), Box<dyn Error>> {
     let theme = Theme::default();
 
     snapshot.transfers().enumerate().for_each(|(idx, balance)| {
-        info!("{} balance #{}", theme.action("Generating"), idx);
+        info!(
+            "{} balance #{} = {:?}",
+            theme.action("Generating"),
+            idx,
+            balance.notes
+        );
```

Comment:
I'm in favour of previous implementation (without the notes debug)
IIRC this was added just to debug